app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/stackTrace/stackTrace.jsx

```diff
@@ -0,0 +1,183 @@
+import { NoItemMessage } from 'components/main/noItemMessage';
+import styles from './stackTrace.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  messageRefCaption: {
+    id: 'StackTrace.messageRefCaption',
+    defaultMessage: 'Go to stack trace in log message',
+  },
+  noStackTrace: {
+    id: 'StackTrace.noStackTrace',
+    defaultMessage: 'No stack trace to display',
+  },
+});
+
+@connect((state) => ({
+  logItemId: activeLogIdSelector(state),
+  projectId: activeProjectSelector(state),
+  pagination: logPaginationSelector(state),
+  pageProperties: pagePropertiesSelector(state, NAMESPACE),
+  logItems: logItemsSelector(state),
+}))
+@injectIntl
+export class StackTrace extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    projectId: PropTypes.string.isRequired,
+    logItemId: PropTypes.string.isRequired,
+    onChangePage: PropTypes.func.isRequired,
+    pagination: PropTypes.object.isRequired,
+    pageProperties: PropTypes.object.isRequired,
+    logItems: PropTypes.array.isRequired,
+  };
+
+  static checkIfStackTraceItemOnThisPage = (logItems, stackTraceItem) =>
+    !!logItems.find((item) => item.id === stackTraceItem.id);
+
+  static getDerivedStateFromProps(props, state) {
+    return state.activePage !== props.pagination.number
+      ? {
+          activePage: props.pagination.number,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            props.logItems,
+            state.stackTraceItem,
+          ),
+        }
+      : null;
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      stackTraceItem: null,
+      isStackTraceItemOnThisPage: false,
+      activePage: props.pagination.number,
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchStackTrace();
+  }
+
+  fetchStackTrace = () => {
+    const { logItems, projectId, logItemId } = this.props;
+    this.setState({
+      loading: true,
+    });
+    fetch(URLS.logItemStackTrace(projectId, logItemId))
+      .then(({ content }) => {
+        const stackTraceItem = (content.length && content[0]) || null;
+        this.setState({
+          stackTraceItem,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            logItems,
+            stackTraceItem,
+          ),
+          loading: false,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          loading: false,
+        });
+      });
+  };
+
+  goToMessageInLog = ({ wasClicked, targetPageNumber }) => {
+    const { isStackTraceItemOnThisPage, stackTraceItem, activePage } = this.state;
+
+    if (isStackTraceItemOnThisPage) {
+      const targetItem = document.getElementById(stackTraceItem.id);
+      targetItem.scrollIntoView({ behavior: 'smooth' });
+      let highlightElement = document.getElementById('highlight');
+      if (!highlightElement) {
+        highlightElement = document.createElement('div');
```

Comment:
@AmsterGet, @DiscoElevator there must be a strong reason for use native js api in react.

Reply:
Let's think how we can highlight table rows without adding/removing DOM nodes directly.

Reply:
Fixed. Now `div` with `highlight` `id` is created in `LogsGrid`. So in this function, we simply manage the necessary classes.